src/jit/lclvars.h

```diff
@@ -650,3 +650,101 @@ class LclVarDsc
 #endif // DEBUG
 
 }; // class LclVarDsc
+
+class LclVarTable final
+{
+    unsigned m_count;
+    unsigned m_capacity;
+    LclVarDsc* m_array;
+
+    void Grow(Compiler& compiler, unsigned newCapacity);
+    void GrowIfNecessary(Compiler& compiler, unsigned requiredCapacity);
+
+public:
+    class Iterator final
+    {
+        friend class LclVarTable;
+
+        LclVarDsc* m_lclVarDsc;
+
+        Iterator(LclVarDsc* lclVarDsc)
+            : m_lclVarDsc(lclVarDsc)
+        {
+        }
+
+    public:
+        inline LclVarDsc* operator*()
+        {
+            return m_lclVarDsc;
+        }
+
+        inline LclVarDsc* operator->()
+        {
+            return m_lclVarDsc;
+        }
+
+        inline bool operator==(const Iterator& other) const
+        {
+            return m_lclVarDsc == other.m_lclVarDsc;
+        }
+
+        inline bool operator!=(const Iterator& other) const
+        {
+            return m_lclVarDsc != other.m_lclVarDsc;
+        }
+
+        inline Iterator& operator++()
+        {
+            ++m_lclVarDsc;
+            return *this;
+        }
+    };
+
+    LclVarTable();
+    LclVarTable(Compiler& compiler, unsigned initialSize);
+
+    LclVarTable& operator=(const LclVarTable& other);
```

Comment:
Why can it be a move assignment? As I see we only use it to init from constructor or from inline lvaTable, that we don't use after.